lib/devicetrust/testenv/testenv.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testenv
+
+import (
+	"context"
+	"net"
+	"time"
+
+	"github.com/gravitational/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+
+	devicepb "github.com/gravitational/teleport/api/gen/proto/go/teleport/devicetrust/v1"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+// E is an integrated test environment for device trust.
+type E struct {
+	DevicesClient devicepb.DeviceTrustServiceClient
+
+	closers []func() error
+}
+
+// Close tears down the test environment.
+func (e *E) Close() error {
+	var errs []error
+	for i := len(e.closers) - 1; i >= 0; i-- {
+		if err := e.closers[i](); err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return trace.NewAggregate(errs...)
+}
+
+// MustNew creates a new E or panics.
+// Callers are required to defer e.Close() to release test resources.
+func MustNew() *E {
```

Comment:
Why not take a `*testing.T`, fail the test on error and add a cleanup as appropriate? Are we ever going to use this outside of a go test environment?

Reply:
Force of habit - I prefer writing "normal Go" to "testing Go" as much as I can. Same results in the end.